Model/Sync/Checkout/Data.php

```diff
@@ -242,32 +242,27 @@ public function prepareLineItems($quote)
         }
         try {
             foreach ($quote->getAllVisibleItems() as $item) {
-                try {
-                    $itemRuleIds = explode(',', $item->getAppliedRuleIds());
-                    if ($ruleId != null || !in_array($ruleId, $itemRuleIds)) {
-                        $couponCode = null;
-                    }
-                    $product = $this->prepareProductObject($item);
+                $itemRuleIds = explode(',', $item->getAppliedRuleIds());
```

Comment:
Why did you decide to keep the outer try-catch and not the inside one?

Reply:
I think that we cannot skip products creation, as Checkout without products is half information, If we are okay with not having all the products synced, I will keep the inner condition. WDYT?

Reply:
The outer try-catch sounds good.